py/dml/codegen.py

```diff
@@ -543,7 +543,12 @@ def expr_unop(tree, location, scope):
             report(WSIZEOFTYPE(tree.site))
             return codegen_sizeof(
                 tree.site, mkLit(tree.site, cident(var), None))
-    rh = codegen_expression_maybe_nonvalue(rh_ast, location, scope)
+    try:
+        rh = codegen_expression_maybe_nonvalue(rh_ast, location, scope)
+    except EIDENT as e:
+        if op == 'sizeof':
```

Comment:
Perhaps add a precondition that e.ident is the name of a type?

Reply:
Are you suggesting running `eval_type`? Because `typedefs` wouldn't cover e.g. `uint32`. I'd really rather not.

Reply:
I would suggest `types.parse_type`. I think it does exactly what you want, but perhaps I missed some corner.

Reply:
`parse_type` alone isn't enough, but when _combined_ with typedefs... maybe. I'm not 100% confident it covers all cases, but I feel like any remaining corner would be ill-advised in the first place.